File: PTA/BasicLevel/Python3/B1002_AC.py
num2hanzi = {'1': 'yi', '2': 'er', '3': 'san', '4': 'si',
             '5': 'wu', '6': 'liu', '7': 'qi', '8': 'ba', '9': 'jiu', '0': 'ling'}

digits = list(map(lambda x: int(x), input()))

digits_sum = 0
for digit in digits:
    digits_sum += digit

num_strs = str(digits_sum)

if len(num_strs) > 0:
    print(num2hanzi[num_strs[0]], end='')

for n in num_strs[1:]:
    print(' ' + num2hanzi[n], end='')

File: PTA/PAT_B/Cpp11/B1002_AC.cc
#include <cstdlib>
#include <iostream>
#include <string>
#include <vector>
using namespace std;
int main() {
  vector<string> num2hanzi{"ling", "yi",  "er", "san", "si",
                           "wu",   "liu", "qi", "ba",  "jiu"};
  string num;
  cin >> num;
  int sum = 0;

  while (not num.empty()) {
    sum += (num.back() - '0');
    num.pop_back();
  }

  vector<string> result;
  while (sum != 0) {
    result.push_back(num2hanzi[sum % 10]);
    sum /= 10;
  }

  cout << result.back();
  result.pop_back();
  for (auto it = result.rbegin(); it < result.rend(); ++it) {
    cout << " " << *it;
  }

  return 0;
}